src/requests/utils.py

```diff
@@ -134,7 +134,10 @@ def super_len(o):
     total_length = None
     current_position = 0
 
-    if hasattr(o, "__len__"):
+    if isinstance(o, str):
```

Comment:
More simply this can be an if without changing the following conditions and just set `o = o.encode(...)` in the block then let the rest of the logic work. As it will hit another condition which will get it's length

Reply:
Sounds fine to me. My usual workflow would be to rewrite my commit and do a force push to my branch (in my fork). Is this ok or would you prefer a separate commit for the change?

Reply:
That's my workflow as well. Feel free to do that